shell/platform/embedder/tests/embedder_unittests.cc

```diff
@@ -1265,11 +1265,163 @@ TEST_F(EmbedderTest, CanDeinitializeAnEngine) {
+  EmbedderConfigBuilder builder(context);
+  builder.SetSoftwareRendererConfig();
+  builder.SetDartEntrypoint("window_metrics_event_all_view_ids");
+
+  fml::AutoResetWaitableEvent ready_latch, message_latch;
+  context.AddNativeCallback(
+      "SignalNativeTest",
+      CREATE_NATIVE_ENTRY(
+          [&ready_latch](Dart_NativeArguments args) { ready_latch.Signal(); }));
+
+  std::string message;
+  context.AddNativeCallback("SignalNativeMessage",
+                            CREATE_NATIVE_ENTRY([&](Dart_NativeArguments args) {
+                              message =
+                                  tonic::DartConverter<std::string>::FromDart(
+                                      Dart_GetNativeArgument(args, 0));
+                              message_latch.Signal();
+                            }));
+
+  auto engine = builder.LaunchEngine();
+  ASSERT_TRUE(engine.is_valid());
+
+  ready_latch.Wait();
+
+  FlutterWindowMetricsEvent metrics = {};
+  metrics.struct_size = sizeof(FlutterWindowMetricsEvent);
+  metrics.width = 800;
+  metrics.height = 600;
+  metrics.pixel_ratio = 1.0;
+  metrics.view_id = 123;
+
+  FlutterAddViewInfo info = {};
+  info.struct_size = sizeof(FlutterAddViewInfo);
+  info.view_id = 123;
+  info.view_metrics = &metrics;
+  info.add_view_callback = [](const FlutterAddViewResult* result) {
+    ASSERT_TRUE(result->added);
+  };
+  ASSERT_EQ(FlutterEngineAddView(engine.get(), &info), kSuccess);
+  message_latch.Wait();
+  ASSERT_EQ("View IDs: [0, 123]", message);
+}
+
+//------------------------------------------------------------------------------
+/// Test that adding a view schedules a frame.
+///
+TEST_F(EmbedderTest, AddViewSchedulesFrame) {
+  auto& context = GetEmbedderContext(EmbedderTestContextType::kSoftwareContext);
+  EmbedderConfigBuilder builder(context);
+  builder.SetSoftwareRendererConfig();
+  builder.SetDartEntrypoint("add_view_schedules_frame");
+  fml::AutoResetWaitableEvent latch;
+  context.AddNativeCallback(
+      "SignalNativeTest",
+      CREATE_NATIVE_ENTRY(
+          [&latch](Dart_NativeArguments args) { latch.Signal(); }));
+
+  fml::AutoResetWaitableEvent check_latch;
+  context.AddNativeCallback(
+      "SignalNativeCount",
+      CREATE_NATIVE_ENTRY(
+          [&check_latch](Dart_NativeArguments args) { check_latch.Signal(); }));
+
+  auto engine = builder.LaunchEngine();
+  ASSERT_TRUE(engine.is_valid());
+
+  // Wait for the application to attach the listener.
+  latch.Wait();
+
+  FlutterWindowMetricsEvent metrics = {};
+  metrics.struct_size = sizeof(FlutterWindowMetricsEvent);
+  metrics.width = 800;
+  metrics.height = 600;
+  metrics.pixel_ratio = 1.0;
+  metrics.view_id = 123;
+
+  FlutterAddViewInfo info = {};
+  info.struct_size = sizeof(FlutterAddViewInfo);
+  info.view_id = 123;
+  info.view_metrics = &metrics;
+  info.add_view_callback = [](const FlutterAddViewResult* result) {
+    ASSERT_TRUE(result->added);
```

Comment:
nit: Here and in a few places below -- given that this is just checking a result (rather than necessary for further checks below) should this be EXPECT_TRUE rather than ASSERT_TRUE? Agreed the two others should be asserts in this case.

Reply:
Ah good point. I did a pass through the tests and "downgraded" checks that could be expects.